### ISSUE TITLE
Case counts validation fails despite passing Python side.

### ISSUE THREAD

**Issue by nnoll** [neherlab/covid19_scenarios#456]
# 🐛 Bug Report

I noticed this while working on scenarios.py but got worried it was a more general problem. Nominally we validate case_counts.json on the data directory side so are confident what we output is correct. However, I don't see this behavior. Instead, when I generate a new case_counts.json file I see no errors and then when the application tries to load this in I get a crash due to a validation error with the case_counts.

This is most likely related to Python dates vs datetimes. Quicktype's python classes utilize the datetime object and thus the output string will include times. I believe the YAML file only specifies Dates and thus leads to an inconsistency.

I was able to solve the scenarios.py problem by enforcing my child class to be typed with Dates but this is ultimately a workaround due to the inconsistency in the output of quicktype.

## How to reproduce

Steps to reproduce the issue:

 1. Generate a new case_counts.json file
 2. Try to open the app
 3. View bug

## 😯 Current Behavior

Validation error

## 🤔 Expected Behavior

No validation error

## 💁 Possible Solution

Either generalize the object in the schema or we need to figure out why Quicktype is doing this.

## 🔦 Context

I was trying to update our data

**Comment by nnoll** [comment]
I can fix the Python side if we decide to keep times as dates only. However, I will hold off and wait for discussion.

**Comment by noleti** [comment]
I believe this is partially related to a transient bug in the ITA data. The parser generated .tsv with date-time format, which I fixed yesterday as part of the schema discussion. As the old .tsv is still in the repo (I normally leave updating data/case-counts to @rneher), you will currently generate .json from those incorrect .tsv, including a wrong date-time. 

I agree that it is surprising that the validate() I added does not catch this during `generate_data.py --output-cases foo.json`. The schema defines 'date' for the 'time' key, which should not include the time according to https://json-schema.org/understanding-json-schema/reference/string.html#dates-and-times . But I can also set 'time' to 'foobar' in a test, and validate() will not complain!
